libs/client/WebhookClient.lua

```diff
@@ -0,0 +1,274 @@
+	if webhook then
+		for key, value in pairs(webhook) do
+			if type(value) ~= "table" then
+				self._options[key] = value
+			end
+		end
+	end
+
+	if options then
+		self:modify(options)
+		for k, v in pairs(options) do
+			self._options[k] = v
+		end
+	end
+end
+
+for name, level in pairs(logLevel) do
+	WebhookClient[name] = function(self, fmt, ...)
+		local msg = self._logger:log(level, fmt, ...)
+		if #self._listeners[name] > 0 then
+			return self:emit(name, msg or format(fmt, ...))
+		end
+	end
+end
+
+-- Getters
+function get.avatar(self)
+	return self._options.avatar
+end
+
+function get.channelId(self)
+	return self._options.channel_id
+end
+
+function get.guildId(self)
+	return self._options.guildId
+end
+
+function get.id(self)
+	return self._options.id
+end
+
+function get.name(self)
+	return self._options.name
+end
+
+function get.token(self)
+	return self._options.token
+end
+
+-- Setters and Modifiers
+function WebhookClient:modify(tbl) 
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
+
+	for k, v in pairs(tbl) do
+		if k ~= "channelId" then -- You can not change the channelId
+			self._options[k] = v
+		else
+			self:info("modify does not accept a channelId key.")
+		end
+	end
+
+	local data, err = self._api:modifyWebhookWithToken(self._options.id, self._options.token, tbl)
+	if data then 
+		return true
+	else
+		return false, err
+	end
+end
+
+function WebhookClient:setAvatar(avatar)
+	avatar = Resolver.base64(avatar) or resolveImage(avatar) or null
+
+	local data, err = self:modify({avatar = avatar})
+	if data then
+		return true
+	else
+		return nil, err
+	end
+end
+
+function WebhookClient:setChannelId(channelId)
+	local data, err = self:modify({channel_id = channelId})
+	if data then
+		return true
+	else
+		return nil, err
+	end
+end
+
+function WebhookClient:setName(name)
+	local data, err = self:modify({name = name})
+	if data then
+		return true
+	else
+		return nil, err
+	end
+end
+
+-- Rest of functions
+function WebhookClient:send(content, options) -- return message
+	local files
+	local query = {
+		wait = options and options.wait or false
+	}
+
+	if options then
+		for key, value in pairs(options) do
+			self._options[key] = value -- merge new options
+		end
+
+		if options.file then -- Only one file can be sent at the same time.
+			cntn, err = parseFile(options.file)
+			if err then
+				return nil, err
+			end
+			files = cntn
+		end
+	end
+
+	if files then
+		self._options.file = files
+	end
+
+	local webhookData = self._options
+	local data, err = self._api:executeWebhook(self._options.id, self._options.token, { -- We insert these values manually so that we don't send extra (irrevelant) information
+		content = content,
+		username = webhookData.name,
+		avatar_url = webhookData.avatarURL,
+		tts = webhookData.tts,
+		embeds = webhookData.embeds
+	}, webhookData.file, wait)
+
+	if data then
+		return true
+	else
+		return nil, err
+	end
+end
+
+function WebhookClient:delete()
```

Comment:
Not necessary to have two methods. Just use `delete`. Kind of a bad name, though. It looks like you're deleting the Client.

Reply:
If you were going to delete the client I would use ``destroy``. Delete is used to delete the webhook from discord.

Reply:
Maybe `deleteWebhook` to indicate that you are actually deleting the webhook and not the client. I do not like "destroy".

Reply:
But then `setName` and `setAvatar` become confusing because you're setting the properties of the webhook, not the client. Hmmm.

Reply:
In the end we need to have in mind that we are managing a webhook using an imitation of a Client. I will change ``delete`` to ``deleteWebhook``